src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,181 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String VALID_LOGIN = "Emma";
+    private static final String VALID_PASSWORD = "qa#wd4f_tp";
+    private static final int VALID_AGE = 20;
+    private static RegistrationServiceImpl service;
+    private static StorageDaoImpl storageDao;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        service = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        User user1 = new User();
+        user1.setLogin("Anna");
+        user1.setPassword("qwerty");
+        user1.setAge(18);
+        storageDao.add(user1);
+        User user2 = new User();
+        user2.setLogin("Bob");
+        user2.setPassword("asdfghj");
+        user2.setAge(34);
+        storageDao.add(user2);
+        User user3 = new User();
+        user3.setLogin("Lucy");
+        user3.setPassword("zxcvbnm$");
+        user3.setAge(20);
+        storageDao.add(user3);
+    }
+
+    @Test
+    void validUser_Ok() {
+        user.setLogin(VALID_LOGIN);
+        user.setPassword(VALID_PASSWORD);
+        user.setAge(VALID_AGE);
+        User actual = service.register(user);
+        assertEquals(user, actual);
+        assertTrue(Storage.people.contains(user));
+        assertEquals(4, Storage.people.size());
+    }
+
+    @Test
+    void existedUserByLogin_NotOk() {
+        user.setLogin("Bob");
+        user.setPassword(VALID_PASSWORD);
+        user.setAge(VALID_AGE);
+        assertThrows(RuntimeException.class, () -> {
+            service.register(user);
+        });
```

Comment:
Brackets are redundant. We can write so, but if you cut it will be better)) And the same in other methods